grails-app/services/org/pih/warehouse/auth/AuthService.groovy

```diff
@@ -14,7 +14,32 @@ import org.pih.warehouse.core.User
 
 class AuthService {
 
-    static ThreadLocal<User> currentUser = new ThreadLocal<User>()
-    static ThreadLocal<Location> currentLocation = new ThreadLocal<Location>()
+    private static ThreadLocal<User> threadLocalUser
+    private static ThreadLocal<Location> threadLocalLocation
 
+    void setCurrentUser(User user) {
+        if (!threadLocalUser) {
+            threadLocalUser = new ThreadLocal<User>()
+        }
+        // previous implementation had an explicit get call
+        // threadLocalUser.set(user?.id ? User.get(user.id) : null)
+        threadLocalUser.set(user)
```

Comment:
I'm not sure if we need the `get(id)` call (is it to guard against accidentally referring to user from a different db session?). In my testing, not having it hasn't caused any problems, but if we called `get(id)` (see the original implementation, in red, at [SecurityInterceptor](https://github.com/openboxes/openboxes/pull/3595/files#diff-5072e24fd0145afc6923503785dde77c36cd285094e89513572c0f609e1579b7)) for a specific use case I didn't happen to check, I'd love to know about it.

Reply:
Turns out we did!